### PR TITLE
Kafka topic updater

### LINKED ISSUE #1281: Create microserver, translating old to new datamodel (and optional new to old datamodel)
**Is your feature request related to a problem? Please describe.**
Customers might not immediately update there nodered flows, therefor a service is needed to translate old data into the new model.

Also it might be needed to convert new data into the the old format.

**Describe the solution you'd like**
A new microservice, translating these messages.
It will pub/sub kafka and must prevent messages going in circles.

### PR DESCRIPTION
**Pull request by Sphingobium**
# Description

This microservice subscribes to old topic structures and then publishes the message with the appropriate payload in the new topic structure

Fixes #1281 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [X] messages of all topics expected to initially work in v1 topic structure were received and properly processed and published
- 
# Checklist:

- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding changes to the changelog (if needed).
- [ ] I have made corresponding notices to the upgrade instructions (if needed)

# Changelog changes

Enter here your text for the changelog (or fill it out directly in Notion if you are internal)

### CODE REVIEW

**Review by DaruZero — CHANGES_REQUESTED** [review]

File: deployment/united-manufacturing-hub/values.yaml [hunk 1]
```diff
@@ -2185,6 +2197,38 @@ kafkatoblob:
...
+
+### kafka-topic-updater ###
+
+kafkatopicupdater:
+  image:
+    repository: unitedmanufacturinghub/kafka-topic-updater
+    pullPolicy: IfNotPresent
+  env:
+    # area and productionline are variables inserted into the topic structure to fill up the old structure to the new one
+    area: area
```
Comment (Scarjit) [inline deployment/united-manufacturing-hub/values.yaml:2209]: Change these to DefaultArea, DefaultProductionLine to match our placeholders in factoryinsight

File: deployment/united-manufacturing-hub/values.yaml [hunk 2]
```diff
@@ -2185,6 +2197,38 @@ kafkatoblob:
...
+    productionline: productionLine
+    # these variables are used for raw topics as they didn't have a proper structure beforehand, so there is no customer, location, asset
+    enterprise: enterprise
```
Comment (Scarjit) [inline deployment/united-manufacturing-hub/values.yaml:2212]: default enterprise should be factoryinsight

File: deployment/united-manufacturing-hub/values.yaml [hunk 3]
```diff
@@ -2185,6 +2197,38 @@ kafkatoblob:
...
+    site: site
```
Comment (Scarjit) [inline deployment/united-manufacturing-hub/values.yaml:2213]: testLocation

File: deployment/united-manufacturing-hub/values.yaml [hunk 4]
```diff
@@ -2185,6 +2197,38 @@ kafkatoblob:
...
+    workcell: workcell
```
Comment (Scarjit) [inline deployment/united-manufacturing-hub/values.yaml:2214]: testMachine

File: deployment/united-manufacturing-hub/values.yaml (hunk 1 above)
Comment (Scarjit) [reply deployment/united-manufacturing-hub/values.yaml:2209]: image

File: golang/pkg/datamodel/messageStructs.go [hunk 5]
```diff
@@ -4,100 +4,170 @@ package datamodel
...
-type ScrapCount struct {
-	TimestampMs uint64 `json:"timestamp_ms"`
-	Scrap       uint64 `json:"scrap"`
-}
-
-type Barcode struct {
-	Barcode     string `json:"barcode"`
-	TimestampMs uint64 `json:"timestamp_ms"`
+// Productadd is new datamodel version of Count
+type Productadd struct {
```
Comment (Scarjit) [inline golang/pkg/datamodel/messageStructs.go:15]: ProductAdd

File: golang/pkg/datamodel/messageStructs.go [hunk 6]
```diff
@@ -4,100 +4,170 @@ package datamodel
...
 }
 
+// Deprecated: use Stateactivity instead
 type Activity struct {
 	TimestampMs uint64 `json:"timestamp_ms"`
 	Activity    bool   `json:"activity"`
 }
 
+// Stateactivity is new datamodel type version of Activity
+type Stateactivity struct {
```
Comment (Scarjit) [inline golang/pkg/datamodel/messageStructs.go:31]: StateActivity

File: golang/pkg/datamodel/messageStructs.go [hunk 7]
```diff
@@ -4,100 +4,170 @@ package datamodel
...
+}
+
+// Deprecated: Use Statereason instead
 type DetectedAnomaly struct {
 	DetectedAnomaly string `json:"detectedAnomaly"`
 	TimestampMs     uint64 `json:"timestamp_ms"`
 }
 
+// Statereason is the new datamodel type version of DetectedAnomaly
+type Statereason struct {
```
Comment (Scarjit) [inline golang/pkg/datamodel/messageStructs.go:43]: StateReason (same capitalization for the ones below)

**Review by Scarjit — CHANGES_REQUESTED** [review]

**Review by Scarjit — APPROVED** [review]

**Pull request merged**